File: pkg/api/data/projectassociation.go
package data

type ProjectAssociationDetail struct {
	ID    string `json:"id"`
	Email string `json:"email"`
	Type  string `json:"type"`
}

type ProjectAssociationCreate struct {
	ProjectID string `json:"projectId" binding:"required"`
	Email     string `json:"email" binding:"required,email"`
	Type      string `json:"type" binding:"required"`
}

type ProjectAssociationUpdate struct {
	ID   string `json:"id" binding:"required"`
	Type string `json:"type" binding:"required"`
}

type ProjectAssociationDelete struct {
	ID string `json:"id" binding:"required"`
}

File: frontend/src/models/projectassociation.ts
import { AssociationType } from "../enums";
import { User } from "./user";

export interface ProjectAssociationDetail {
    id: string,
    email: string,
    type: AssociationType,
};

export interface ProjectAssociationCreate {
    projectId: string,
    email: string,
    type: AssociationType
};

export interface ProjectAssociationUpdate {
    id: string,
    type: AssociationType
};

export interface ProjectAssociationDelete {
    id: string
}